### ISSUE TITLE
.NET 3.5 Target

### ISSUE THREAD

**Issue by Trojaner** [Rohansi/Mond#74]
Is there any way to support .NET 3.5 (mono)?

Yes I know this would prob bamboozle the Async/Task support but thats not something *that* important for me.

I plan to integrate this into the RocketMod plugin framework (at https://github.com/RocketMod/Rocket.Scripting) but it targets only .NET 3.5 as some games it supports still use Unity 5.

**Comment by Rohansi** [comment]
I did have it running on Unity 5 before after stripping stuff like Async out. Definitely possible to do but not sure how valuable it would be. I'd expect performance to be pretty bad because `MondValue` is an immutable class and will create a ton of garbage.

It could be switched to a struct (which would break compatibility of course) but performance was much better on regular .NET when using a class because of all the defensive struct copying. The `struct-mondvalue` branch is experimenting with the new `ref` features of C# 7.x to solve both of these. But I'm 99.9% sure Unity 5 won't handle that well.

TLDR: 100% doable, if you add it in I'll merge it, but keep an eye on performance when using with Unity.
